src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -977,6 +995,7 @@ bool ClusterTaskManager::ReleaseCpuResourcesFromUnblockedWorker(
       for (unsigned int i = 0; i < overflow_cpu_instances.size(); i++) {
         RAY_CHECK(overflow_cpu_instances[i] == 0) << "Should not be overflow";
       }
+      worker->MarkBlocked();
```

Comment:
Don't we do this in node_manager anyways if true is returned?

Reply:
that would be because i didn't push the changes to node manager (but yes). 

I think ideally we keep it here so it can be unit tested.